crates/transaction-generator-lib/src/lib.rs

```diff
@@ -306,22 +348,121 @@ pub async fn create_txn_generator_creator(
     )
 }
 
-fn get_account_to_burn_from_pool(
-    accounts_pool: &Arc<RwLock<Vec<LocalAccount>>>,
-    needed: usize,
-) -> Vec<LocalAccount> {
-    let mut accounts_pool = accounts_pool.write();
-    let num_in_pool = accounts_pool.len();
-    if num_in_pool < needed {
+/// Simple object pool structure, that you can add and remove from multiple threads.
+/// Taking is done at random positions, but sequentially.
+/// Overflow replaces at random positions as well.
+///
+/// It's efficient to lock the objects for short time - and replace
+/// in place, but its not a concurrent datastructure.
+pub struct ObjectPool<T> {
+    pool: RwLock<Vec<T>>,
+}
+
+impl<T> ObjectPool<T> {
+    pub(crate) fn new_initial(initial: Vec<T>) -> Self {
+        Self {
+            pool: RwLock::new(initial),
+        }
+    }
+
+    pub(crate) fn new() -> Self {
+        Self::new_initial(Vec::new())
+    }
+
+    pub(crate) fn reserve(&self, additional: usize) {
+        self.pool.write().reserve(additional);
+    }
+
+    pub(crate) fn add_to_pool(&self, mut addition: Vec<T>) {
+        assert!(!addition.is_empty());
```

Comment:
This should be no-op instead of assert ?

Reply:
it could, but in general to catch more bugs early - places that always expect to have something to add, this invariant is good. For other places, having them do the check before calling is fine